app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -151,16 +154,40 @@ class FxaRepo(
  *  A wrapper for [Profile]. This insulates us from any upstream changes to the API, and
  *  allows us to validate related data at the edge of our app.
  *
- *  TODO it looks like some of these fields should be nonnullable (at least [uid] and [email]. We should:
- *  - Reach out to FxA devs for clarification re: which of these fields are actually nullable
- *  - Validate that important fields are nonnull at the edge of our app
- *  - Replace other fields ([avatar], possibly [displayName]) with default values
+ *  Note that [Profile] has additional fields not reflected in [FxaProfile] (namely
+ *  [Profile.uid] and [Profile.email]). These are not currently required by our app, but
+ *  can be added if that ever changes.
  */
 data class FxaProfile(
-    val uid: String?,
-    val email: String?,
-    val avatar: Avatar?,
-    val displayName: String?
+    val avatar: ImageSetStrategy,
+    val displayName: String
 )
 
-fun Profile.toDomainObject() = FxaProfile(this.uid, this.email, this.avatar, this.displayName ?: this.email)
+private const val DEFAULT_FXA_AVATAR_URL = "https://firefoxusercontent.com/00000000000000000000000000000000"
+private const val DEFAULT_AVATAR_RESOURCE = R.drawable.ic_default_avatar
+
+fun Profile.toDomainObject(): FxaProfile {
+    val avatar = this.avatar
```

Comment:
nit: Can't you just refer to avatar as `avatar`? What's the value of setting a temporary variable? Here and below.

Reply:
`Smart cast to 'Avatar' is impossible, because 'this.avatar' is a public API property declared in a different module`

Super annoying, but the compiler can't guarantee that the value isn't mutated between checks, so it makes sense.